### PR TITLE
Optimize topology spreading for null selector

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Avoid checking pods when the selector is null. This way, a spreading constraint with an empty selector can be more useful for disabling spreading for a specific pod.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Another potential solution was to not add the constraint into the PreFilter/PreScore state. However, the constraints have other fields that still need to be checked.

Unit tests pass before and after the change.

#### Does this PR introduce a user-facing change?

```release-note
kube-scheduler: Optimized implementation of null labelSelector in topology spreading.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/podtopologyspread/common.go [hunk 1]
```diff
@@ -145,6 +145,9 @@ func mergeLabelSetWithSelector(matchLabels labels.Set, s labels.Selector) labels
 }
 
 func countPodsMatchSelector(podInfos []*framework.PodInfo, selector labels.Selector, ns string) int {
+	if selector.Empty() {
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/podtopologyspread/common.go:148]: how about exiting here when all constraints have empty selector? https://github.com/kubernetes/kubernetes/blob/1cb334960c4288fc7830a16c97f5f8489f140b5b/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L78

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/podtopologyspread/common.go:148]: that will save iterating over the nodes

Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/podtopologyspread/common.go:148]: yes, but it changes this behavior below:

https://github.com/kubernetes/kubernetes/blob/1cb334960c4288fc7830a16c97f5f8489f140b5b/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L81

Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/podtopologyspread/common.go:148]: Note that if you set a DoNotSchedule constraint, then we also don't apply default constraints in scoring. In that case the list of constraints in PreScore is zero.

**Pull request merged**